fadtk/model_loader.py

```diff
@@ -56,12 +56,11 @@ def _get_embedding(self, audio: np.ndarray):
     def load_wav(self, wav_file: Path):
         wav_data, _ = soundfile.read(wav_file, dtype='int16')
         wav_data = wav_data / 32768.0  # Convert to [-1.0, +1.0]
-        if self.audio_len is not None:
-            if wav_data.shape[0] < self.sr*4:
-                padding_size = self.sr*4 - len(wav_data)
-                wav_data = np.pad(wav_data, (0, padding_size), 'constant', constant_values=(0, 0))
-            elif wav_data.shape[0] > self.sr*4:
-                wav_data = wav_data[:self.sr*4]
+        
+        # Ensure the audio length is correct
+        if self.audio_len is not None and wav_data.shape[0] != self.audio_len * self.sr:
+            raise RuntimeError(f"Audio is too long ({wav_data.shape[0] / self.sr:.2f} seconds > {self.audio_len} seconds)."
+                                + f"\n\t- {wav_file}")
```

Comment:
I changed the role of 'audio_len' from chopping/padding to audio length check. Raises error if audio file has different length.